### PR TITLE
480 data label

### PR DESCRIPTION
**Pull request by jeffwa428**
**Explain the details for making this change. What existing problem does the pull request solve?**
Created new component for data label

**Related github/jira issue (required)**:
Closes https://github.com/infor-design/enterprise-wc/issues/480

**Steps necessary to review your pull request (required)**:
- Go to http://localhost:4300/ids-data-label
- Check top and left positioned data label
- Check data label for RTL (French locale)
- Should be same as https://main-enterprise.demo.design.infor.com/components/form/example-labels.html

**Included in this Pull Request**:
- [x] An e2e or functional test for the bug or feature.
- [] A note to the change log.

### CODE REVIEW

File: src/components/ids-data-label/ids-data-label.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    }
+  }
+
+  get label() { return this.getAttribute(attributes.LABEL); }
+
+  /**
+   * Sets to label
+   * @param {string} value label string
+   */
+  set labelPosition(value) {
```
Comment (tmcconechy) [inline src/components/ids-data-label/ids-data-label.js:67]: I noticed that changing these properties inline in the html does not update. Should fix that. It should add/remove the class and be dynamic for all properties

File: src/components/ids-data-label/ids-data-label.scss [hunk 2]
```diff
@@ -0,0 +1,51 @@
+@import '../../core/ids-base';
+
+:host([lang='fr']) {
```
Comment (tmcconechy) [inline src/components/ids-data-label/ids-data-label.scss:3]: Pls Make sure this works when setting via the container...I dont see that it is or it doesnt update

File: src/core/ids-attributes.js [hunk 3]
```diff
@@ -133,6 +133,7 @@ export const attributes = {
   LABEL_PROGRESS: 'label-progress',
   LABEL_REQUIRED: 'label-required',
   LABEL_TOTAL: 'label-total',
+  LABEL_POSITION: 'label-position',
```
Comment (tmcconechy) [inline src/core/ids-attributes.js:136]: Alphabetically pls

File: test/ids-data-label/ids-data-label-func-test.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    elem.remove();
+    expect(document.querySelectorAll('ids-data-label').length).toEqual(1);
+    expect(errors).not.toHaveBeenCalled();
+  });
+
+  it('renders correctly', () => {
+    expect(dataLabel.outerHTML).toMatchSnapshot();
+  });
+
+  it('renders with label-position', () => {
```
Comment (tmcconechy) [inline test/ids-data-label/ids-data-label-func-test.js:34]: Maybe add tests for making it dynamically change and the colon setter?

**Review by tmcconechy — CHANGES_REQUESTED** [review]
A few suggestions inline. Also try to fix the lint errors on the build (npm run lint)
I didnt see any problems with e2e percy / maybe it was temporary

File: demos/ids-data-label/example.html [hunk 5]
```diff
@@ -0,0 +1,10 @@
+<ids-layout-grid auto="true">
+  <ids-layout-grid-cell>
+    <ids-text font-size="16" type="h1">Data Label</ids-text>
+  </ids-layout-grid-cell>
+</ids-layout-grid>
+<ids-layout-grid auto="true" id="ids-data-label-list" gap="md">
+  <ids-data-label label="Shipping to" label-position="left">Los Angeles, California 90001 USA</ids-data-label>
+  <ids-data-label label="Shipping to" label-position="left" lang="fr">Los Angeles, California 90001 USA</ids-data-label>
```
[the remark this answers is not in the document]
Comment (EdwardCoyle) [reply demos/ids-data-label/example.html:8]: Until we build langauge switch into the demoapp a bit further, maybe put this on a separate test page so we can still test it quickly?

[the remark this answers is not in the document]
Comment (tmcconechy) [reply demos/ids-data-label/example.html:8]: @EdwardCoyle i dont think it needs a seperate page can just use the container on that page. And actually you cant really change language on it like that it needs to be on the container now so i dont think this is correct on a new page..

File: demos/ids-data-label/example.html [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+    <ids-text font-size="12" type="h1">Data Label</ids-text>
+  </ids-layout-grid-cell>
+</ids-layout-grid>
+<ids-layout-grid auto="true" id="ids-data-label-list" gap="md">
+  <ids-data-label label="Shipping to" label-position="left">Los Angeles, California 90001 USA</ids-data-label>
+  <ids-data-label label="Shipping to">Los Angeles, California 90001 USA</ids-data-label>
+</ids-layout-grid>
+
+<ids-layout-grid auto="true">
+  <ids-container language="fr">
```
Comment (tmcconechy) [inline demos/ids-data-label/example.html:12]: Please just use the container at the top of the page..

File: demos/ids-data-label/index.yaml [hunk 7]
```diff
@@ -0,0 +1,9 @@
+---
+ component: Data Label
+ link: ids-data-label
+ description: Readonly form label with data
+ category: Form Inputs
+ keywords: label, data, readonly, read only
```
Comment (tmcconechy) [inline demos/ids-data-label/index.yaml:6]: Add to [example.js (or index.js)](https://github.com/infor-design/enterprise-wc/blob/main/demos/example.js)  so it shows on http://localhost:4300/i

File: src/components/ids-data-label/README.md [hunk 8]
```diff
@@ -0,0 +1,28 @@
+# Ids Data Label Component
+
+## Description
+
+The IDS Data Label component used to show both label and value.
```
Comment (tmcconechy) [inline src/components/ids-data-label/README.md:5]:
```suggestion
The IDS Data Label component used to show both label and value for readonly (non editable data).
```

File: src/components/ids-data-label/README.md [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+
+## Use Cases
+
+Typically, these data labels are useful to show information with top/left/right label positioned.
```
Comment (tmcconechy) [inline src/components/ids-data-label/README.md:9]:
```suggestion
- When you need to show a readonly data item with a label
```

File: src/components/ids-data-label/README.md [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+
+## Feature (With the Code Examples)
+
+```html
```
Comment (tmcconechy) [inline src/components/ids-data-label/README.md:13]:
```suggestion

Shows a data label with one top label and one with a left label.

```html
```

File: src/components/ids-data-label/ids-data-label.js [hunk 11]
```diff
@@ -0,0 +1,111 @@
+/* eslint-disable no-nested-ternary */
```
Comment (tmcconechy) [inline src/components/ids-data-label/ids-data-label.js:1]: Is this needed?

File: src/components/ids-data-label/ids-data-label.js [hunk 12]
```diff
@@ -0,0 +1,111 @@
...
+   * @returns {Array} The propertires in an array
+   */
+  static get attributes() {
+    return [
+      ...super.attributes,
+      attributes.LABEL,
+      attributes.LABEL_POSITION,
+      attributes.MODE,
+      attributes.VERSION,
+      attributes.LANGUAGE
```
Comment (tmcconechy) [inline src/components/ids-data-label/ids-data-label.js:45]: Could be in order

File: src/components/ids-data-label/ids-data-label.scss [hunk 13]
```diff
@@ -0,0 +1,52 @@
+@import '../../core/ids-base';
+
+:host([language='fr']) {
+  .left-positioned {
+    direction: rtl;
```
Comment (tmcconechy) [inline src/components/ids-data-label/ids-data-label.scss:5]: Only ar and hi are RTL so no need to make french RTL. The difference the colon spacing ' : ' vs ': '

File: src/components/ids-data-label/ids-data-label.js [hunk 14]
```diff
@@ -0,0 +1,111 @@
...
+      return `${this.labelPosition}-positioned`;
+    }
+    return 'top-positioned';
+  }
+
+  /**
+   * @returns {string} css class for data-label
+   */
+  get colon() {
+    return this.labelPosition === 'left' ? ':' : '';
```
Comment (tmcconechy) [inline src/components/ids-data-label/ids-data-label.js:103]: The logic should be if french a space on both sides ' : ' vs ': ' for all others.

**Review by tmcconechy — CHANGES_REQUESTED** [review]
Some more things im seeing here. Lets just have 2 in the page not a seperate one for french and we can test it by the container on the top of the page

also can add one more test that 
- starts in en and test the colon is right
- changes to fr and test the colon is right

File: src/components/ids-data-label/ids-data-label.scss [hunk 15]
```diff
@@ -0,0 +1,50 @@
+@import '../../core/ids-base';
+
+:host([language='fr']) {
```
Comment (tmcconechy) [inline src/components/ids-data-label/ids-data-label.scss:3]:
```suggestion
:host([language='fr']) {
```

I noticed that if using language `fr-FR` or `fr-CA` it doesnt work can we add cases for language = 'fr-FR' and 'fr-CA'?

As those are also valid. And im not sure if just changing here is enough?

Comment (EdwardCoyle) [reply src/components/ids-data-label/ids-data-label.scss:3]: Could this work?
```suggestion
:host([language*='fr']) {
```

**Review by deep7102 — COMMENTED** [review]
Looks Nice! @jeffwa428 
saw only one thing when it wraps looks weird (try on smaller view port) might just need to adjust the css `white-space: nowrap;`
image

I try to edit with console this css property and looks as below:
image

**Review by tmcconechy — APPROVED** [review]

**Review by deep7102 — APPROVED** [review]
Thanks! @jeffwa428 
LGTM 🍌

**Pull request merged**